### PR TITLE
Material Flow Simulation Result v3.0.0

### PR DESCRIPTION
**Pull request by dominikoeh**
## Description

 -->

Closes https://github.com/eclipse-tractusx/sldt-semantic-models/issues/726

## Sample Data

```json
{
  "materialFlowScenarioRequest": {
    "scenarioSimResults": {
      "resultOwnId": "916b5688-8bd8-4d7e-83b9-e0d40939274e",
      "resultOwnSimRunInitial": {
        "owner": {
          "bpns": "BPNS00000007OTZ3"
        },
        "dataQuality": 0,
        "description": "Please simulate asap",
        "comment": "successful simulation ",
        "expirationTimestamp": "2023-03-24T09:15:24.000Z",
        "runId": "0fece48b-c8d1-4180-1a9caca6d67e",
        "shipments": [
          {
            "handlingUnits": [
              {
                "name": "Palette",
                "volume": 1,
                "weight": 189,
                "batches": [
                  {
                    "unitOfMeasurement": "KG",
                    "materialName": "KK1000GR-Gehäuse-Rot",
                    "quantity": 50,
                    "materialNumber": "KK1000GR",
                    "materialHazardousGoods": false,
                    "batchSerialNumber": "Batch_1",
                    "batchOrderId": "Order-0001",
                    "batchExpirationTimestamp": "2023-08-22T16:00:00.000Z",
                    "batchNumber": "45"
                  }
                ],
                "handlingUnitId": "HUT_1",
                "amount": 1
              }
            ],
            "shipmentId": "DE51515151",
            "recipientTimestampPlanned": "2023-04-19T09:00:00.000Z",
            "destination": {
              "bpns": "BPNS00000007OTZ3"
            },
            "recipient": {
              "bpns": "BPNS00000007OTZ3"
            },
            "logistics": {
              "bpns": "BPNS00000007OTZ3"
            },
            "preceding": {
              "bpns": "BPNS00000007OTZ3"
            },
            "splittingAllowed": true,
            "destinationTimestamp": "2023-03-19T09:00:00.000Z"
          }
        ],
        "timestamp": "2023-03-09T14:13:42.806Z"
      },
      "resultOwnSimRunUpdated": {
        "owner": {
          "bpns": "BPNS00000007OTZ3"
        },
        "dataQuality": 0,
        "description": "Please simulate asap",
        "comment": "successful simulation ",
        "expirationTimestamp": "2023-03-24T09:15:24.000Z",
        "runId": "0fece48b-c8d1-4180-1a9caca6d67e",
        "shipments": [
          {
            "handlingUnits": [
              {
                "name": "Palette",
                "volume": 1,
                "weight": 189,
                "batches": [
                  {
                    "unitOfMeasurement": "KG",
                    "materialName": "KK1000GR-Gehäuse-Rot",
                    "quantity": 50,
                    "materialNumber": "KK1000GR",
                    "materialHazardousGoods": false,
                    "batchSerialNumber": "Batch_1",
                    "batchOrderId": "Order-0001",
                    "batchExpirationTimestamp": "2023-08-22T16:00:00.000Z",
                    "batchNumber": "45"
                  }
                ],
                "handlingUnitId": "HUT_1",
                "amount": 1
              }
            ],
            "shipmentId": "DE51515151",
            "recipientTimestampPlanned": "2023-04-19T09:00:00.000Z",
            "destination": {
              "bpns": "BPNS00000007OTZ3"
            },
            "recipient": {
              "bpns": "BPNS00000007OTZ3"
            },
            "logistics": {
              "bpns": "BPNS00000007OTZ3"
            },
            "preceding": {
              "bpns": "BPNS00000007OTZ3"
            },
            "splittingAllowed": true,
            "destinationTimestamp": "2023-03-19T09:00:00.000Z"
          }
        ],
        "timestamp": "2023-03-09T14:13:42.806Z"
      }
    },
    "scenarioParameter": [
      {
        "unitOfMeasurement": "KG",
        "parameterComment": "updated Delivery Date",
        "materialName": "KK1000GR-Gehäuse-Rot",
        "parameterQuantityUpdated": 1,
        "parameterId": "847c71e5-614a-468b-a3a0-674bf2af3004",
        "materialNumber": "KK1000GR",
        "parameterDeliveryDateUpdated": "2023-10-10T09:00:00.000Z",
        "parameterDeliveryDateInitial": "2023-10-09T10:00:00.000Z",
        "parameterOrderId": "OID-011123546",
        "parameterQuantityInitial": 1
      }
    ],
    "scenarioHeader": {
      "scenarioOwnerRole": "Customer",
      "scenarioCreationTimestamp": "2023-10-04T09:10:00.000Z",
      "scenarioExpirationTimestamp": "2023-10-07T09:10:00.000Z",
      "scenarioOwner": {
        "bpns": "BPNS00000007OTZ3"
      },
      "scenarioDescription": "Changes in Delivery Date",
      "scenarioId": "8d464b8b-6977-4952-8a22-0489067ca081",
      "scenarioTitle": "Delivery Modification"
    }
  }
}
```

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.6.0)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by catroest** [comment]
Please look at the models:
- serial part
- batch
- business partner number
- quantity

for external references as I see here a lot of attributes re-emergence.

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+:OwnerCharacteristic a samm-c:SingleEntity ;
+   samm:preferredName "Owner Characteristic"@en ;
+   samm:description "Site identifier of the simulation originator"@en ;
+   samm:dataType :BusinessPartnerSite .
+
+:DataQualityOptions a samm-c:Enumeration ;
+   samm:preferredName "Data Quality Options"@en ;
+   samm:description "Possible values determining data quality"@en ;
+   samm:dataType xsd:positiveInteger ;
```
Comment (agg3fe) [inline io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl:108]: here also change to xsd:nonNegativeInteger

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+   samm-c:values ( "0"^^xsd:positiveInteger "1"^^xsd:positiveInteger "2"^^xsd:positiveInteger "3"^^xsd:positiveInteger "4"^^xsd:positiveInteger "5"^^xsd:positiveInteger ) .
```
Comment (agg3fe) [inline io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl:109]: xsd:nonNegativeInteger

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+   samm:exampleValue "2023-08-22T16:00:00.000Z"^^xsd:dateTime .
+
+:quantity a samm:Property ;
+   samm:preferredName "Quantity"@en ;
+   samm:description "Amount of materials"@en ;
+   samm:characteristic :QuantityCharacteristic ;
+   samm:exampleValue "50"^^xsd:float .
+
+:unitOfMeasurement a samm:Property ;
+   samm:preferredName "UnitOfMeasurement"@en ;
```
Comment (agg3fe) [inline io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl:311]: UnitOfMeasurement should not be together

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+:owner a samm:Property ;
+   samm:preferredName "Owner"@en ;
+   samm:description "Site identifier of the simulation originator"@en ;
+   samm:characteristic :OwnerCharacteristic .
+
+:dataQuality a samm:Property ;
+   samm:preferredName "Data Quality"@en ;
+   samm:description "Information on quality of simulation results"@en ;
+   samm:characteristic :DataQualityOptions ;
+   samm:exampleValue "0"^^xsd:positiveInteger .
```
[the remark this answers is not in the document]
Comment (dominikoeh) [reply io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl]: type has been changed

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl (hunk 2 above)
Comment (dominikoeh) [reply io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl]: type has been changed

File: io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl (hunk 3 above)
Comment (dominikoeh) [reply io.catenax.material_flow_simulation_result/3.0.0/MaterialFlowSimulationResultAspect.ttl]: pefered name has been changed

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.material_flow_simulation_result/RELEASE_NOTES.md [hunk 5]
```diff
@@ -2,6 +2,12 @@
 
 All notable changes to this model will be documented in this file.
 
+## [3.0.0] 2024-05-27
```
Comment (agg3fe) [inline io.catenax.material_flow_simulation_result/RELEASE_NOTES.md:5]: please change the date to 21st

**Comment by agg3fe** [comment]
@dominikoeh Let me know when the changes are done.

**Comment by dominikoeh** [comment]
> Please look at the models:
> 
> * serial part
> * batch
> * business partner number
> * quantity
> 
> for external references as I see here a lot of attributes re-emergence.

@agg3fe 

- Adopted BPNS reference from shared BPN Aspect.
- Other data types seem less suitable due to different structures or meanings:
  - Serial part: Not all required parameters are covered.
  - Batch: Contains unnecessary information for simulation.
  - Quantity: Requires specific type assignment (piece, weight, volume), less flexible than current model.

**Comment by catroest** [comment]
> > Please look at the models:
> > 
> > * serial part
> > * batch
> > * business partner number
> > * quantity
> > 
> > for external references as I see here a lot of attributes re-emergence.
> 
> @agg3fe
> 
> * Adopted BPNS reference from shared BPN Aspect.
> * Other data types seem less suitable due to different structures or meanings:
>   
>   * Serial part: Not all required parameters are covered.
>   * Batch: Contains unnecessary information for simulation.
>   * Quantity: Requires specific type assignment (piece, weight, volume), less flexible than current model.
> 
> The same applies to Material Flow Scenario Request. The BPN trait is re-used in that aspect model by a shared super structure

Agreed on most parts.
But in quantity you can choose the unit of a specific type. So weight can be for example kilogram, gram, pound and so on.

**Comment by tobias-kutzler-iff** [comment]
> > > Please look at the models:
> > > 
> > > * serial part
> > > * batch
> > > * business partner number
> > > * quantity
> > > 
> > > for external references as I see here a lot of attributes re-emergence.
> > 
> > 
> > @agg3fe
> > 
> > * Adopted BPNS reference from shared BPN Aspect.
> > * Other data types seem less suitable due to different structures or meanings:
> >   
> >   * Serial part: Not all required parameters are covered.
> >   * Batch: Contains unnecessary information for simulation.
> >   * Quantity: Requires specific type assignment (piece, weight, volume), less flexible than current model.
> > 
> > The same applies to Material Flow Scenario Request. The BPN trait is re-used in that aspect model by a shared super structure
> 
> Agreed on most parts. But in quantity you can choose the unit of a specific type. So weight can be for example kilogram, gram, pound and so on.

The aspect model "quantity" offers different properties where the type is directly assigned and specified. To use this aspect it is necessary to choose specifically a property dependent on the type. But this approach is not generic as we have implemented it.

Short example:
We provide the possibility to define a value and in a separate property the selection of the unit (e.g. liter, piece, kg). Only two properties and filled with a value are used.
When using the aspect model "quantity" or its properties on the one hand it is necessary to statically specify the type of the value or it is necessary to add for each used type the appropriate property. But then only one of the defined properties would contain a value and all other properties would be empty.

Based on this argumentation we want to reject the use of the aspect "quantity" but we will keep it in mind for further developments.

**Pull request merged**